### PR TITLE
User authentication feature

### PR DESCRIPTION
**Pull request by jshidalgokommit**
features implemented: 

a. Log in and log out. 

b. Create, edit, delete my blog posts.

c. Add comments in other blog posts only for logged users.

### CODE REVIEW

**Comment by jshidalgokommit** [comment]
Hi @JoseCastroA, the new authentication structure was implemented. Now table Users store auth information and user data (name, last name)

File: app/assets/javascripts/devise/registrations/edit.js [hunk 1]
```diff
@@ -0,0 +1,9 @@
+/**
+ * show confirm dialog and cancel user account
+ */
+function cancel_account(){
+    if (confirm('are you sure?')){
+        document.getElementById("cancel_account_form").submit();
+    }
+}
+
```
Comment (JoseCastroA) [inline app/assets/javascripts/devise/registrations/edit.js:9]: @jshidalgokommit Please remove this file

File: app/controllers/articles_controller.rb [hunk 2]
```diff
@@ -27,7 +28,7 @@ def create
   end
 
   def edit
-    @article = Article.find(params[:id])
+    @article
```
Comment (JoseCastroA) [inline app/controllers/articles_controller.rb:31]: @jshidalgokommit If you do` @article = Article.find(params[:id])` on the `verify_owner` function, you don't need this line

File: app/models/comment.rb [hunk 3]
```diff
@@ -4,6 +4,7 @@ class Comment < ApplicationRecord
   include Visible
 
   belongs_to :article
+  belongs_to :user
 
-  validates :commenter, :body, presence: true
+  validates :commenter, :body, :user_id, presence: true
```
Comment (JoseCastroA) [inline app/models/comment.rb:9]: Do we need to validate the presence of `article_id`? @jshidalgokommit

**Review by JoseCastroA — CHANGES_REQUESTED** [review]

**Comment by jshidalgokommit** [comment]
Hi @JoseCastroA, the requested changes were implemented

**Review by JoseCastroA — APPROVED** [review]

**Pull request merged**